test/deploy-pipeline-mkp-cli.sh

```diff
@@ -0,0 +1,208 @@
+echo "add service account for running the test workflow"
+kubectl create serviceaccount ${ARGO_KSA} -n ${NAMESPACE} --dry-run -o yaml | kubectl apply -f -
+kubectl create clusterrolebinding test-admin-binding --clusterrole=cluster-admin --serviceaccount=${NAMESPACE}:${ARGO_KSA} --dry-run -o yaml | kubectl apply -f -
+
+if [ "$ENABLE_WORKLOAD_IDENTITY" = true ]; then
+  ARGO_GSA="test-argo"
+  # Util library including create_gsa_if_not_present and bind_gsa_and_ksa functions.
+  source "$DIR/../manifests/kustomize/wi-utils.sh"
+  create_gsa_if_not_present $ARGO_GSA
+
+  gcloud projects add-iam-policy-binding $PROJECT \
+    --member="serviceAccount:$ARGO_GSA@$PROJECT.iam.gserviceaccount.com" \
+    --role="roles/editor" \
+    > /dev/null # hide verbose output
+  bind_gsa_and_ksa $ARGO_GSA $ARGO_KSA $PROJECT $NAMESPACE
+
+  verify_workload_identity_binding $ARGO_KSA $NAMESPACE
+fi
+
+echo "=================START MKP DEPLOY================"
+
+GCR_IMAGE_TAG=${GCR_IMAGE_TAG:-latest}
+
+# Configure gcloud as a Docker credential helper
+gcloud auth configure-docker
+
+# Delete argo first because KFP comes with argo too
+kubectl delete namespace argo --wait || echo "No argo installed"
+
+# Install the application resource definition
+kubectl apply -f "https://raw.githubusercontent.com/GoogleCloudPlatform/marketplace-k8s-app-tools/master/crd/app-crd.yaml"
+
+# Grant user ability for using Role-Based Access Control
+kubectl create clusterrolebinding cluster-admin-binding \
+  --clusterrole cluster-admin \
+  --user $(gcloud config get-value account)
+
+export APP_INSTANCE_NAME=kubeflow-pipelines-test
+
+# Install mpdev
+BIN_FILE="$HOME/bin/mpdev"
+if ! which mpdev; then
+  echo "Install mpdev"
+  docker pull gcr.io/cloud-marketplace-tools/k8s/dev
+  mkdir -p $HOME/bin/
+  touch $BIN_FILE
+  export PATH=$HOME/bin:$PATH
+  docker run gcr.io/cloud-marketplace-staging/marketplace-k8s-app-tools/k8s/dev:remove-ui-ownerrefs cat /scripts/dev > "$BIN_FILE"
+  chmod +x "$BIN_FILE"
+fi
+
+export MARKETPLACE_TOOLS_TAG=remove-ui-ownerrefs
+export MARKETPLACE_TOOLS_IMAGE=gcr.io/cloud-marketplace-staging/marketplace-k8s-app-tools/k8s/dev
+
+KFP_MANIFEST_DIR=${DIR}/../manifests/gcp_marketplace
+pushd ${KFP_MANIFEST_DIR}
+# Update the version value on schema.yaml and application.yaml
+sed -ri 's/publishedVersion:.*/publishedVersion: '"$GCR_IMAGE_TAG"'/' schema.yaml
+sed -ri 's/version:.*/version: '"$GCR_IMAGE_TAG"'/' ./chart/kubeflow-pipelines/templates/application.yaml
+
+# Build new deployer
+export REGISTRY=gcr.io/$(gcloud config get-value project | tr ':' '/')
+export APP_NAME=$COMMIT_SHA
+export DEPLOYER_NAME=$REGISTRY/$APP_NAME/deployer
+
+docker build --tag $DEPLOYER_NAME -f deployer/Dockerfile .
+
+docker push $DEPLOYER_NAME:$GCR_IMAGE_TAG
+
+# Copy rest images and rename current images
+export MKP_VERSION=$GCR_IMAGE_TAG
+export GCR_FOLDER=$REGISTRY/$APP_NAME
+
+echo MKP_VERSION:$MKP_VERSION, GCR_FOLDER:$GCR_FOLDER
+
+export ARGO_VERSION=v2.3.0-license-compliance
```

Comment:
One idea I thought of, extract MKP release related config from https://github.com/kubeflow/pipelines/blob/master/.release.cloudbuild.yaml into a separate config, so that the extracted config can be parameterized to also get used in this test.

WDYT?

Reply:
That sounds like a good idea. One question, where shall we put the MKP config? It seems like yaml does not support import/include. Doe Cloud build workflow support retrieving parameters from config files?

Reply:
We can configure two cloud build triggers on release instead, no need to let one import another.

I think you can add parameters with default values suitable for release. And in tests, we override them to reuse the config.

Anyway, I don't feel these are a blocker to this PR. It could be a separate improvement, because this is already working.